### PR TITLE
Workflow for labeled iOS PRs in 4.x branch

### PR DESCRIPTION
**Pull request by asenyaev**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: .github/workflows/PR-4.x.yaml [hunk 1]
```diff
@@ -21,5 +22,9 @@ jobs:
   macOS-X64:
     uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-macOS-x86_64.yaml@main
 
+  iOS:
+    if: contains(github.event.pull_request.labels.*.name, 'ios')
```
Comment (asenyaev) [inline .github/workflows/PR-4.x.yaml:26]: Seems `platform: ios/osx` label cannot be found using this condition. But if to use just `ios`, it works.

File: .github/workflows/PR-4.x.yaml [hunk 2]
```diff
@@ -2,6 +2,7 @@ name: PR:4.x
 
 on:
   pull_request:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
```
Comment (asenyaev) [inline .github/workflows/PR-4.x.yaml:5]: > By default, a workflow only runs when a pull_request event's activity type is `opened`, `synchronize`, or `reopened`

I guess we want to re-run jobs, if labels were changed or added by a maintainer.

Comment (asenyaev) [reply .github/workflows/PR-4.x.yaml:5]: But there is an issue when PR is created with labels. It runs several workflows (`opened` and `labeled` worked). However, sometimes PR are created without any labels.

If to keep the default statement and add labels after some period of time for a PR, need to re-commit, because a workflow won't be triggered in this case.

**Review by fengyuentau — APPROVED** [review]
LGTM 👍

**Comment by asenyaev** [comment]
Fixed parsing a label `platform: ios/osx`.

File: .github/workflows/PR-4.x.yaml (hunk 2 above)
Comment (asenyaev) [reply .github/workflows/PR-4.x.yaml]: Removed this line accordingly to https://github.com/opencv/opencv/pull/22112#discussion_r899325558.

File: .github/workflows/PR-4.x.yaml (hunk 1 above)
Comment (asenyaev) [reply .github/workflows/PR-4.x.yaml]: Fixed it.

File: .github/workflows/PR-4.x.yaml [hunk 3]
```diff
@@ -21,5 +21,9 @@ jobs:
   macOS-X64:
     uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-macOS-x86_64.yaml@main
 
+  iOS:
+    if: "${{ contains(github.event.pull_request.labels.*.name, 'platform: ios/osx') }}"
```
Comment (alalek) [inline .github/workflows/PR-4.x.yaml:25]: Do we really need this condition?
(BB-based CI always run iOS builds (in some reduced set))

Is there an option to move and handle that in workflows repository?

Comment (asenyaev) [reply .github/workflows/PR-4.x.yaml:25]: What does "BB-based CI" mean?

Yes, there is an option, but if to compare with `khadas-vim3-tests` job, we run iOS workflow in CI repository, what means that we should have a large `if` condition there to handle all cases.

Comment (asenyaev) [reply .github/workflows/PR-4.x.yaml]: Removed this line to run iOS Framework build on each commit in a PR.

File: apps/traincascade/boost.cpp [hunk 4]
```diff
@@ -1677,7 +1677,7 @@ void CvCascadeBoost::write( FileStorage &fs, const Mat& featureMap ) const
     fs << CC_WEAK_CLASSIFIERS << "[";
     for( int wi = 0; wi < weak->total; wi++)
     {
-        /*sprintf( cmnt, "tree %i", wi );
+        /*snprintf( cmnt, sizeof(cmnt), "tree %i", wi );
```
Comment (alalek) [inline apps/traincascade/boost.cpp:1680]: These changes are not related to this PR.
Please rebase to have 1 commit only.

Comment (asenyaev) [reply apps/traincascade/boost.cpp]: Done.

**Review by alalek — APPROVED** [review]
